enterprise/cmd/frontend/internal/registry/allow.go

```diff
@@ -23,24 +23,41 @@ func init() {
 		return false
 	}
 
+	frontendregistry.IsRemoteExtensionPublisherAllowed = func(p registry.Publisher) bool {
+		if getAllowOnlySourcegraphAuthoredExtensionsFromSiteConfig() {
+			return isSourcegraphAuthoredExtension(p)
+		}
+
+		return true
+	}
+
 	frontendregistry.FilterRemoteExtensions = func(extensions []*registry.Extension) []*registry.Extension {
+		var keep []*registry.Extension
+
 		allowedExtensions := getAllowedExtensionsFromSiteConfig()
-		if allowedExtensions == nil {
-			// Default is to allow all extensions.
-			return extensions
+		if allowedExtensions != nil {
+			allow := make(map[string]any)
```

Comment:
```suggestion
			allow := make(map[string]struct{})
```

Not directly related to this PR but based on the actual usage, `any` is not actually needed, because `struct{}` would allocate zero space. Let's fix it since you touched the line 😁